.github/workflows/build-macos.yml

```diff
@@ -0,0 +1,81 @@
+---
+# CI code for build and test on macOS
+name: macOS Build and test
+on:
+  push: # Master branch checks only validate the build and generate artifacts for testing.
+    branches:
+      - master
+  pull_request: null # PR checks only validate the build and generate artifacts for testing.
+  workflow_dispatch: # Dispatch runs build and validate, then push to the appropriate storage location.
+    inputs:          # Inputs for this workflow can be omitted
+      type:
+        description: Build Type
+        default: nightly
+        required: true
+      version:
+        description: Version Tag
+        default: nightly
+        required: true
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+jobs:
+  build-test:
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix:
+        include:
+          - name: macos-12
+            runner: macos-12
+          - name: macos-13
+            runner: macos-13
+          - name: macos-14-M1
+            runner: macos-14
+    steps:
+      - uses: actions/checkout@v4
+        id: checkout
+        with:
+          submodules: recursive
+      - name: Setup fundamental depedencies
+        id: setup-runner
+        run: |
+          brew install bash
+      - name: Install netdata dependencies
+        id: install-nd-dep
+        run: |
+          export DISABLE_TELEMETRY=1
```

Comment:
This should be set in an `environment` block like we do at the top of the checks workflow. The only reason it isn’t in the build workflow that you copied for this is that all the builds there run in containers and thus don’t inherit the runner environment.

Reply:
k